src/beliefpropagation/beliefpropagation.jl

```diff
@@ -1,104 +1,67 @@
 function message_tensors(
-  tn::ITensorNetwork;
-  nvertices_per_partition=nothing,
-  npartitions=nothing,
-  subgraph_vertices=nothing,
-  kwargs...,
+  ptn::PartitionedGraph; itensor_constructor=inds_e -> ITensor[dense(delta(inds_e))]
 )
-  return message_tensors(
-    partition(tn; nvertices_per_partition, npartitions, subgraph_vertices); kwargs...
-  )
-end
-
-function message_tensors_skeleton(subgraphs::DataGraph)
-  mts = DataGraph{vertextype(subgraphs),vertex_data_type(subgraphs),ITensorNetwork}(
-    directed_graph(underlying_graph(subgraphs))
-  )
-  for v in vertices(mts)
-    mts[v] = subgraphs[v]
-  end
-
-  return mts
-end
-
-function message_tensors(
-  subgraphs::DataGraph;
-  itensor_constructor=inds_e -> ITensor[dense(delta(i)) for i in inds_e],
-)
-  mts = message_tensors_skeleton(subgraphs)
-  for e in edges(subgraphs)
-    inds_e = commoninds(subgraphs[src(e)], subgraphs[dst(e)])
-    itensors = itensor_constructor(inds_e)
-    mts[e] = ITensorNetwork(itensors)
-    mts[reverse(e)] = dag(mts[e])
+  mts = Dict()
+  for e in partitionedges(ptn)
+    src_e_itn = subgraph(ptn, src(e))
+    dst_e_itn = subgraph(ptn, dst(e))
+    inds_e = commoninds(src_e_itn, dst_e_itn)
+    mts[e] = itensor_constructor(inds_e)
+    mts[reverse(e)] = dag.(mts[e])
   end
   return mts
 end
 
 """
-DO a single update of a message tensor using the current subgraph and the incoming mts
+Do a single update of a message tensor using the current subgraph and the incoming mts
 """
 function update_message_tensor(
-  tn::ITensorNetwork,
-  subgraph_vertices::Vector,
-  mts::Vector{ITensorNetwork};
-  contract_kwargs=(; alg="density_matrix", output_structure=path_graph_structure, maxdim=1),
+  ptn::PartitionedGraph, edge::PartitionEdge, mts; contract_kwargs=(; alg="exact")
```

Comment:
I would define `default_contract_kwargs() = (; alg="exact")` and then set `contract_kwargs=default_contract_kwargs()`, then you can use that throughout the code in other parts and we can easily change defaults if we want to from one location.